params/config.go

```diff
@@ -49,52 +57,67 @@ var (
+		EIP150Block:                  big.NewInt(0),
+		EIP150Hash:                   common.HexToHash("0x62e0fde86e34c263e250fbcd5ca4598ba8ca10a1d166c8526bb127e10b313311"),
+		EIP155Block:                  big.NewInt(10),
+		EIP158Block:                  big.NewInt(10),
+		ByzantiumBlock:               big.NewInt(1700000),
+		ConstantinopleBlock:          nil,
+		TIP2019Block:                 big.NewInt(0),
+		TIPSigningBlock:              big.NewInt(0),
+		TIPRandomizeBlock:            big.NewInt(0),
+		BlackListHFBlock:             big.NewInt(0),
+		TIPTomoXCancellationFeeBlock: big.NewInt(0),
+		TIPTRC21FeeBlock:             big.NewInt(0),
+		Ethash:                       new(EthashConfig),
 	}
 
 	// AllEthashProtocolChanges contains every protocol change (EIPs) introduced
 	// and accepted by the Ethereum core developers into the Ethash consensus.
 	//
 	// This configuration is intentionally not using keyed fields to force anyone
 	// adding flags to the config to also have to set these fields.
-	AllEthashProtocolChanges = &ChainConfig{big.NewInt(1337), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, new(EthashConfig), nil, nil}
+	AllEthashProtocolChanges = &ChainConfig{
+		ChainId:                      big.NewInt(1337),
+		HomesteadBlock:               big.NewInt(0),
+		EIP150Block:                  big.NewInt(0),
+		EIP155Block:                  big.NewInt(0),
+		EIP158Block:                  big.NewInt(0),
+		ByzantiumBlock:               big.NewInt(0),
+		TIPTomoXCancellationFeeBlock: big.NewInt(0),
+		Ethash:                       new(EthashConfig),
+	}
 
 	// AllPosvProtocolChanges contains every protocol change (EIPs) introduced
 	// and accepted by the Ethereum core developers into the Posv consensus.
 	//
 	// This configuration is intentionally not using keyed fields to force anyone
 	// adding flags to the config to also have to set these fields.
-	AllPosvProtocolChanges   = &ChainConfig{big.NewInt(89), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, nil, nil, &PosvConfig{Period: 0, Epoch: 30000}}
-	AllCliqueProtocolChanges = &ChainConfig{big.NewInt(1337), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, nil, &CliqueConfig{Period: 0, Epoch: 30000}, nil}
-	TestChainConfig          = &ChainConfig{big.NewInt(1), big.NewInt(0), nil, false, big.NewInt(0), common.Hash{}, big.NewInt(0), big.NewInt(0), big.NewInt(0), nil, new(EthashConfig), nil, nil}
-	TestRules                = TestChainConfig.Rules(new(big.Int))
+	AllPosvProtocolChanges = &ChainConfig{
+		ChainId:        big.NewInt(89),
+		HomesteadBlock: big.NewInt(0),
+		EIP150Block:    big.NewInt(0),
+		EIP155Block:    big.NewInt(0),
+		EIP158Block:    big.NewInt(0),
+		ByzantiumBlock: big.NewInt(0),
+		Posv:           &PosvConfig{Period: 0, Epoch: 30000},
+	}
+
+	TestChainConfig = &ChainConfig{
```

Comment:
As my understanding, TestChainConfig is used for our testnet. Why all hardfork blocks aren't pre-defined here. Please clarify.

Reply:
`TestChainConfig` is used for unit testing only. `TestnetChainConfig` above is the one which is used for the testnet.

Reply:
I'll add comment for this. And we will need to choose the testnet fork blocks before this PR is merged. I suggest the final `TestnetChainConfig` as below
```go
TestnetChainConfig = &ChainConfig{
		ChainId:                      big.NewInt(89),
		HomesteadBlock:               big.NewInt(0),
		DAOForkBlock:                 nil,
		DAOForkSupport:               true,
		EIP150Block:                  big.NewInt(0),
		EIP150Hash:                   common.HexToHash("0x62e0fde86e34c263e250fbcd5ca4598ba8ca10a1d166c8526bb127e10b313311"),
		EIP155Block:                  big.NewInt(0),
		EIP158Block:                  big.NewInt(0),
		ByzantiumBlock:               big.NewInt(0),
		ConstantinopleBlock:          nil,
		TIP2019Block:                 big.NewInt(0),
		TIPSigningBlock:              big.NewInt(0),
		TIPRandomizeBlock:            big.NewInt(0),
		BlackListHFBlock:             big.NewInt(0),
		TIPTomoXCancellationFeeBlock: big.NewInt(0),
		TIPTRC21FeeBlock:             big.NewInt(0),
		Posv: &PosvConfig{
			Period:              2,
			Epoch:               900,
			Reward:              250,
			RewardCheckpoint:    900,
			Gap:                 5,
			FoudationWalletAddr: common.HexToAddress("0x0000000000000000000000000000000000000068"),
		},
	}
```

Reply:
We also need to define the default bootnodes for the testnet here either
https://github.com/BuildOnViction/tomochain/blob/master/params%2Fbootnodes.go#L41